ambry-replication/src/main/java/com/github/ambry/replication/ReplicaThread.java

```diff
@@ -639,14 +644,31 @@ private ReplicaMetadataResponse getReplicaMetadataResponse(List<RemoteReplicaInf
   }
 
   /**
-   * Gets the missing store messages by comparing the messages from the remote node
+   * Determines if {@link MessageInfo} container in the status of DELETED_IN_PROGRESS or INACTIVE.
+   * @param messageInfo A message info class that contains basic info about a blob
+   * @return {@code true} if the blob associates with the deprecated container, {@code false} otherwise.
+   * Deprecated containers status include DELETE_IN_PROGRESS and INACTIVE.
+   */
+  private boolean isDeprecatedContainer(MessageInfo messageInfo) {
```

Comment:
While the logic is correct, it seems a bit heavy handed to pass AccountService to ReplicaThread and see all this logic here (which probably also is similar in other places).  I would suggest adding a generic interface that lets the replica thread know if the messageInfo should be included or skipped.  The interface would have one method like:
boolean skipFromReplication(MessageInfo messageInfo);
Then pass this interface to ReplicaThread instead of AccountService.  In the future, we may want to extend the logic of what messages to skip and have that be transparent to replication.

Reply:
Thanks for the suggestion. I use Predicate interface and implement a ReplicationSkipPredicate to check what message need to be skipped.